src/ir/stack-utils.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2020 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// stack-utils.h: Utilities for manipulating and analyzing stack machine code in
+// the form of Stacky IR.
+//
+// Stacky IR represents stack machine code using normal Binaryen IR types by
+// imposing the following structural constraints:
+//
+//  1. Function bodies and children of control flow (except If conditions) must
+//     be blocks.
+//
+//  2. Blocks may have any Expressions except for Pops as children. The sequence
+//     of instructions in a block follows the same validation rules as in
+//     WebAssembly. That means that any expression may have a concrete type, not
+//     just the final expression in the block.
+//
+//  3. All other children must be Pops, which are used to determine the input
```

Comment:
I kind of like "Poppy" as a name because of this. But up to you...

Reply:
Drive-by comment: I like stuff with weird but perfectly cromulent names. Embiggens its meaning.

Reply:
Sounds good :) Since folks like the name, I will use it in the code as well. Using one name consistently will certainly simplify discussion, too.